rllib/core/testing/tf/bc_module.py

```diff
@@ -100,3 +101,78 @@ def from_model_config(
         }
 
         return cls(**config)
+
+
+class BCTfRLModuleWithSharedGlobalEncoder(TfRLModule):
+    def __init__(self, encoder, local_dim, hidden_dim, action_dim):
+        super().__init__()
+
+        self.encoder = encoder
+        self.policy_head = tf.keras.Sequential(
+            [
+                tf.keras.layers.Dense(
+                    hidden_dim + local_dim,
+                    input_shape=(hidden_dim + local_dim,),
+                    activation="relu",
+                ),
+                tf.keras.layers.Dense(hidden_dim, activation="relu"),
+                tf.keras.layers.Dense(action_dim),
+            ]
+        )
+
+    def input_specs_inference(self):
+        return self._common_input_spec()
+
+    def input_specs_exploration(self):
+        return self._common_input_spec()
+
+    def input_specs_train(self):
+        return self._common_input_spec()
```

Comment:
dude, can these be the defaults? like all 3 input_specs_xxx() functions can _common_input_spec() in the base class, so we don't have to write these all the time?

Reply:
The default is different module by module, right? 
For example here, we have global and local under obs. How can this go to base class? You can leave the specs empty if you don't feel like spec checking :-))

Reply:
The default is different module by module, right? 
For example here, we have global and local under obs. How can this go to base class? You can leave the specs empty if you don't feel like spec checking :-))

Reply:
The default is different module by module, right? 
For example here, we have global and local under obs. How can this go to base class? You can leave the specs empty if you don't feel like spec checking :-))

Reply:
The default is different module by module, right? 
For example here, we have global and local under obs. How can this go to base class? You can leave the specs empty if you don't feel like spec checking :-))

Reply:
oh sorry, what I meant was you have
```
def input_specs_train(self):
        return self._common_input_spec()
```
in the base class. for each different module, people then just override ``_common_input_spec()`` once.
the pattern of all three input_specs_xxx sharing the same spec should be common?

Reply:
I think for specs it makes sense.